src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,53 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private T[] elements;
+    private int size;
+    private static final int INITIAL_CAPACITY = 10;
+
+    @SuppressWarnings("unchecked")
+    public MateStack() {
+        elements = (T[]) new Object[INITIAL_CAPACITY];
+        size = 0;
+    }
 
+    public void push(T value) {
+        if (size == elements.length) {
+            resize();
+        }
+        elements[size++] = value;
     }
```

Comment:
Code Quality Issue: It's recommended to encapsulate the logic of adding an element to the array in a separate method. This will make the code more readable and maintainable.